services/brig/test/integration/API/Search.hs

```diff
@@ -263,15 +263,15 @@ testOrderName brig = do
   nameMatch <- userQualifiedId <$> createUser' True searchedWord brig
   namePrefixMatch <- userQualifiedId <$> createUser' True (searchedWord <> "suffix") brig
   refreshIndex brig
-  results <- searchResults <$> executeSearch brig searcher searchedWord
-  let resultUIds = map contactQualifiedId results
   let expectedOrder = [nameMatch, namePrefixMatch]
-  let dbg = "results: " <> show results <> "\nsearchedWord: " <> cs searchedWord
+  -- possibly there is some delay in refreshing of the index writes being
+  -- visible for subsequent reads, so we try this a few times.
+  results <- aFewTimes 12 (toIds <$> executeSearch brig searcher searchedWord) (== expectedOrder)
   liftIO $
     assertEqual
-      ("Expected order: name match, name prefix match.\n\nSince this test fails sporadically for unknown reasons here is some debug info:\n" <> dbg)
+      "If this test fails again, maybe we should consider deleting it and living with some more uncertainty"
       expectedOrder
-      resultUIds
+      results
```

Comment:
This won't actually fix the test. The problem is in the production code. So, if you really want to fix it by running again, you should generate  the name again.